### PR TITLE
Add Nuxt 3 + Appwrite web app

### PR DESCRIPTION
**Pull request by EshaanAgg**
## What does this PR do?

Adds a demo for Nuxt 3 + Appwrite so that the developer community has get started with Appwrite more easily in future!
You can checkout the app [here](https://github.com/eshaanagg/workout-buddy)!

## Related PRs and Issues

* https://github.com/appwrite/appwrite/issues/4224

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -313,6 +314,9 @@ Share your apps here! Submit a pull request!
 
 * **Tailwinder** This an open-source complete community website project created for Appwrite hackathon. [Source Code](https://github.com/MooseSaeed/Tailwinder) [Demo](https://dev.to/moose_said/introducing-tailwinder-home-for-tailwind-fans-powered-with-appwrite-4c0o) [Video Demo](https://vimeo.com/manage/videos/709162501)
 
+### Nuxt
+* **Appwrite + Nuxt 3 Workout-Buddy** This is a Nuxt 3 App that uses Appwrite as a backend to showcase the uses of Appwrite database and collections, Authentication and Pinia stores through CRD operations!  [Source Code](https://github.com/EshaanAgg/workout-buddy)
```
Comment (stnguyen90) [inline README.md:318]: Minor corrections:

```suggestion
* **Appwrite + Nuxt 3 Workout-Buddy** This is a Nuxt 3 App that uses Appwrite as a backend to showcase the uses of Appwrite Databases and Collections, Authentication, and Pinia stores through CRUD operations! [Source Code](https://github.com/EshaanAgg/workout-buddy)
```

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
This app is awesome! 🤯 Amazing work! Just a few comments:

1. Could you please use the Appwrite CLI to pull the collections and you can include the appwrite.json with your code?
2. I noticed you have a createdAt attribute in your collection. Why not use the out of the box `$createdAt`?
3. Typically, Appwrite apps have session persistence so when a user comes back to the app, they don't need to log in again. That doesn't seem to be working for your app. Would you be able to get that working?

**Comment by EshaanAgg** [comment]
Sure! I'll try to implement these changes!

**Comment by EshaanAgg** [comment]
@stnguyen90 
Added the requested changes.
1. Used Appwrite CLI to create `appwrite.json`, and included the same in the repository as well the README.md of the project.
2. Deleted the redundant `createdAt` attribute from the`workouts` collection and started using the in-built `$createdAt` attribute for documents.
3. Added support for persistence of users (removed the auth middleware, and implemented the same in the `navbar` component by the means of redirection: did this to avoid conflicts with the Nuxt 3 life cycle)

**Review by stnguyen90 — COMMENTED** [review]
@EshaanAgg, something weird seems to be happening now...Even after logging out, I'm directed to the /workouts page instead of the login/signup page when browsing to the home page. Is that expected?

**Comment by EshaanAgg** [comment]
@stnguyen90 
I can't seem to replicate the issue. For me, the redirection is to `/login` page after clicking on the `logout` button.
image
This is the code for the `handleLogout` function which ensures the same.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
> For me, the redirection is to `/login` page after clicking on the `logout` button

@EshaanAgg and then what happens after you browse to the root (ie. `http://localhost:3000/`) after logging out?

Few other things...

in `nuxt.config.ts`, nuxt doesn't like the `toString()`:

```
databaseID: process.env.NUXT_APP_DATABASE_ID.toString(),
```

I also have this error. do you?

```
Nuxt 3.0.0-rc.11 with Nitro 0.5.4                                                                                                                       17:16:58
                                                                                                                                                        17:16:58
  > Local:    http://localhost:3000/ 
  > Network:  http://192.168.1.43:3000/

✔ Nitro built in 195 ms                                                                                                                           nitro 17:16:59
ℹ Vite client warmed up in 818ms                                                                                                                        17:16:59
Error getting Account TypeError: __vite_ssr_import_1__.Client is not a constructor
    at Object.provider (/Users/steven/Projects/hacktoberfest/eshaanagg-workout-buddy/api/index.js:15:20)
    at Object.getAccount (/Users/steven/Projects/hacktoberfest/eshaanagg-workout-buddy/api/index.js:39:16)
    at Proxy.fetchAccount (/Users/steven/Projects/hacktoberfest/eshaanagg-workout-buddy/stores/account.js:30:61)
    at Proxy.<anonymous> (file:///Users/steven/Projects/hacktoberfest/eshaanagg-workout-buddy/node_modules/pinia/dist/pinia.mjs:1365:30)
    at /Users/steven/Projects/hacktoberfest/eshaanagg-workout-buddy/components/Navbar.vue:21:83
    at Module.withAsyncContext (/Users/steven/Projects/hacktoberfest/eshaanagg-workout-buddy/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:7497:21)
    at setup (/Users/steven/Projects/hacktoberfest/eshaanagg-workout-buddy/components/Navbar.vue:21:47)
    at _sfc_main.setup (/Users/steven/Projects/hacktoberfest/eshaanagg-workout-buddy/components/Navbar.vue:125:23)
    at callWithErrorHandling (/Users/steven/Projects/hacktoberfest/eshaanagg-workout-buddy/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:157:22)
    at setupStatefulComponent (/Users/steven/Projects/hacktoberfest/eshaanagg-workout-buddy/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:7118:29)
```

**Comment by EshaanAgg** [comment]
@stnguyen90 
Pushed a new commit.
Since I wasn't able to figure out the reason for the error, I decided to remove redirection from the home page, and instead created buttons in the navbar that enable user to navigate on the basis of the fact if there are logged in or not, and the current page they are on.

**Comment by stnguyen90** [comment]
> I tried browsing to / after logging out, and it redirected me to /workouts instead of /login.

Shouldn't it redirect you to login?

> Even I am getting the same error logged in my console, but I can't seem to fathom the reason for the same.

Did you try rebuilding without SSR?

**Review by stnguyen90 — COMMENTED** [review]
> Even I am getting the same error logged in my console, but I can't seem to fathom the reason for the same.

Did you try rebuilding without SSR?

**Comment by EshaanAgg** [comment]
> Did you try rebuilding without SSR?

I did but no avail.

**Comment by stnguyen90** [comment]
@EshaanAgg seems to work for me:

image

**Review by stnguyen90 — APPROVED** [review]
Hey 👋 awesome work on your PR! We've approved your work and it'll be merged soon!

**Comment by EshaanAgg** [comment]
Thanks so much @stnguyen90!
Can't explain how much I appreciate you taking out the time to review my code multiple times!

**Comment by stnguyen90** [comment]
Thank YOU! For your hard work on the app!

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**